lib/galaxy/tools/__init__.py

```diff
@@ -1981,6 +1980,13 @@ def handle_input(
             collection_info=collection_info,
             completed_jobs=completed_jobs,
         )
+
+        tags = incoming.get("tags", [])
+        if tags:
```

Comment:
From a backend perspective, Looks good! Just a small suggestion, maybe check if tags is a list to avoid issues if the input isn’t as expected.
```suggestion
        if isinstance(tags, list) and tags:
```